torch/csrc/jit/codegen/cuda/index_compute.cpp

```diff
@@ -918,6 +970,15 @@ generateIndexAndExtentMap(
     loops.pop_back();
   }
 
+  if (tv->definition()->isA<WelfordOp>()) {
```

Comment:
Does this assume `WelfordOp` is the only expression type with multiple outputs? If so, would it be possible to generalize it so that it could work with any future expressions with multiple outputs?

Reply:
Yes. `WelfordOp` is currently the only multi-output use case we considered so far. This PR was trying to support `WelfordOp` with minimal generalizations but if we have other multi-output cases we can generalize. 

On the index compute side the implementation is more temporary than architectural due to the limitation that the loop variable now can only be mapped to one of the outputs. This part will be re-factored after we switch to index compute based on local compute-at and domain maps (@csarofeen). I'd prefer adding multi-output support at that point if we do decide to generalize.

Reply:
I think it's okay to start with something specific to Welford and generalize it later, as long as it is guarded with assertion about the assumption. For example, if something is only meant to work with Welford, then it should be preceded by TORCH_INTERNAL_ASSERT.

Reply:
Added TORCH_INTERNAL_ASSERT for the multioutput case. Thanks.